src/extension.ts

```diff
@@ -72,7 +73,22 @@ type CMakeToolsQueryMapFn = (cmt: CMakeTools) => Thenable<string | string[] | nu
  */
 class ExtensionManager implements vscode.Disposable {
   constructor(public readonly extensionContext: vscode.ExtensionContext) {
-    telemetry.activate();
+    telemetry.activate(extensionContext);
+    this.showCMakeLists = new Promise<boolean>(resolve => {
+      const experimentationService: Promise<IExperimentationService | undefined> | undefined = telemetry.getExperimentationService();
+      if (experimentationService) {
+        void experimentationService
+              .then(expSrv => expSrv!.getTreatmentVariableAsync<boolean>("vscode", "partialActivation_showCMakeLists"))
+              .then(showCMakeLists => {
+                if (showCMakeLists !== undefined) {
+                  resolve(showCMakeLists);
+                } else {
+                  resolve(false);
+                }
+              });
+      }
```

Comment:
something's wrong with our linter settings. It didn't catch the missing resolve for the "else" case. :(